### PR TITLE
DEV-1688: Migrate to GitHub Workflows

### PR DESCRIPTION
**Pull request by m-terel**

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+          version: v1.30
+          args: -c .golangci.yml
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Test
+        run: go test -v ./...
```
Comment (breathbath) [inline .github/workflows/go.yml:25]: Could you add -race flag here?

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+name: Go
+on: [push]
+jobs:
+  golangci:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (breathbath) [inline .github/workflows/go.yml:8]: Why not fetch depth 0 too?

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+      - name: go mod vendor
```
Comment (breathbath) [inline .github/workflows/go.yml:10]: Can you check if golangci-lint won’t analyze the vendor folder?

Comment (breathbath) [inline .github/workflows/go.yml:10]: Can you check if golangci-lint won’t analyze the vendor folder?

Comment (m-terel) [reply .github/workflows/go.yml:10]: It seems by documentation it's skipped by default. Please see `skip-dirs-use-default` https://golangci-lint.run/usage/configuration/

File: .github/workflows/go.yml (hunk 1 above)
Comment (m-terel) [reply .github/workflows/go.yml]: Done.

File: .github/workflows/go.yml (hunk 3 above)
Comment (m-terel) [reply .github/workflows/go.yml:10]: Duplicate comment.

File: .github/workflows/go.yml (hunk 2 above)
Comment (m-terel) [reply .github/workflows/go.yml:8]: Sorry, why do we need to fetch all history for all branches and tags here? I thought it should be done just for a release that is specified in https://goreleaser.com/ci/actions/#workflow

Comment (breathbath) [reply .github/workflows/go.yml:8]: Doesn’t 0 means skip history?

Comment (breathbath) [reply .github/workflows/go.yml:8]: Doesn’t 0 means skip history?

**Review by breathbath — APPROVED** [review]

**Pull request merged**